universal-application-tool-0.0.1/app/views/admin/questions/QuestionEditView.java

```diff
@@ -342,66 +326,12 @@ private SelectWithLabel repeaterOptions(
 
   private QuestionForm getFreshQuestionForm(QuestionType questionType)
       throws UnsupportedQuestionTypeException {
-    QuestionForm questionForm;
-    switch (questionType) {
-      case ADDRESS:
-        questionForm = new AddressQuestionForm();
-        break;
-      case CHECKBOX:
-        questionForm = new CheckboxQuestionForm();
-        break;
-      case DROPDOWN:
-        questionForm = new DropdownQuestionForm();
-        break;
-      case FILEUPLOAD:
-        questionForm = new FileUploadQuestionForm();
-        break;
-      case NAME:
-        questionForm = new NameQuestionForm();
-        break;
-      case NUMBER:
-        questionForm = new NumberQuestionForm();
-        break;
-      case RADIO_BUTTON:
-        questionForm = new RadioButtonQuestionForm();
-        break;
-      case REPEATER:
-        questionForm = new RepeaterQuestionForm();
-        break;
-      case TEXT:
-        questionForm = new TextQuestionForm();
-        break;
-      default:
-        throw new UnsupportedQuestionTypeException(questionType);
-    }
-    return questionForm;
+    return QuestionFormBuilder.create(questionType);
```

Comment:
suggestion: We can pretty cleanly get rid of `getFreshQuestionForm` and `getQuestionFormFromQuestionDefinition` now and just replace calls to them with calls to `QuestionFormBuilder.create`

Reply:
Good suggestion. Done.